plugins/output/flight/flight.go

```diff
@@ -44,38 +45,112 @@ type Flight struct {
+		// For example, JSON does not define types and therefore parses all integers
+		// as floats. If the schema expects an integer in the field, Apache Arrow will panic.
+		for i, schemaField := range schemaFields {
+			switch schemaField.Type.ID() {
+			case arrow.TIMESTAMP:
+				builder.Field(i).(*array.TimestampBuilder).AppendValues([]arrow.Timestamp{arrow.Timestamp(timeInt)}, nil)
+			case arrow.STRING:
+				metricTag := getTag(metric, schemaField, i)
+				builder.Field(i).(*array.StringBuilder).AppendValues([]string{metricTag}, nil)
+			case arrow.INT32:
+				metricField := getField(metric, schemaField, i)
+				switch value := metricField.(type) {
+				case int32:
+					builder.Field(i).(*array.Int32Builder).AppendValues([]int32{value}, nil)
+				case int64:
+					builder.Field(i).(*array.Int32Builder).AppendValues([]int32{int32(value)}, nil)
+				case float32:
+					builder.Field(i).(*array.Int32Builder).AppendValues([]int32{int32(value)}, nil)
+				case float64:
+					builder.Field(i).(*array.Int32Builder).AppendValues([]int32{int32(value)}, nil)
+				}
+			case arrow.INT64:
+				metricField := getField(metric, schemaField, i)
+				switch value := metricField.(type) {
+				case int32:
+					builder.Field(i).(*array.Int64Builder).AppendValues([]int64{int64(value)}, nil)
+				case int64:
+					builder.Field(i).(*array.Int64Builder).AppendValues([]int64{value}, nil)
+				case float32:
+					builder.Field(i).(*array.Int64Builder).AppendValues([]int64{int64(value)}, nil)
+				case float64:
+					builder.Field(i).(*array.Int64Builder).AppendValues([]int64{int64(value)}, nil)
+				}
+			case arrow.FLOAT32:
+				metricField := getField(metric, schemaField, i)
+				switch value := metricField.(type) {
+				case int32:
+					builder.Field(i).(*array.Float32Builder).AppendValues([]float32{float32(value)}, nil)
+				case int64:
+					builder.Field(i).(*array.Float32Builder).AppendValues([]float32{float32(value)}, nil)
+				case float32:
+					builder.Field(i).(*array.Float32Builder).AppendValues([]float32{value}, nil)
+				case float64:
+					builder.Field(i).(*array.Float32Builder).AppendValues([]float32{float32(value)}, nil)
+				}
+			case arrow.FLOAT64:
+				metricField := getField(metric, schemaField, i)
+				switch value := metricField.(type) {
+				case int32:
+					builder.Field(i).(*array.Float64Builder).AppendValues([]float64{float64(value)}, nil)
+				case int64:
+					builder.Field(i).(*array.Float64Builder).AppendValues([]float64{float64(value)}, nil)
+				case float32:
+					builder.Field(i).(*array.Float64Builder).AppendValues([]float64{float64(value)}, nil)
+				case float64:
+					builder.Field(i).(*array.Float64Builder).AppendValues([]float64{value}, nil)
+				}
+			}
+		}
 	}
 
+	// Create a new Record from the RecordBuilder.
 	rec := builder.NewRecord()
 	defer rec.Release()
 
 	return f.writer.Write(rec)
 }
 
+// getTag returns the value of the metric tag with the name equal to the name of the schema field.
```

Comment:
Maybe just "Return the value of the metric tag with the name equal to the name of the schema field."